lints/lint_rsa_signature_algo_parameter_not_null.go

```diff
@@ -0,0 +1,118 @@
+	return c.PublicKeyAlgorithmOID.Equal(util.OidRSAEncryption)
+}
+
+func (l *rsaEncryptionParamNotNULL) Execute(c *x509.Certificate) *LintResult {
+	// Try byte comparison for Algorithm
+	// Determine offset (SEQUENCE + length) first. Notably 1024 encodes length in 2 bytes, 2048 and up in 3 bytes
+	offset := getOffset(c.RawSubjectPublicKeyInfo)
+	if len(c.RawSubjectPublicKeyInfo) <= len(expectedSPKIAlgoBytes)+offset {
+		return &LintResult{Status: Fatal, Details: "error malformed pkixPublicKey"}
+	}
+
+	if bytes.Compare(c.RawSubjectPublicKeyInfo[offset:len(expectedSPKIAlgoBytes)+offset], expectedSPKIAlgoBytes) == 0 {
+		return &LintResult{Status: Pass}
+	}
+
+	input := cryptobyte.String(c.RawSubjectPublicKeyInfo)
+
+	var publicKeyInfo cryptobyte.String
+	if !input.ReadASN1(&publicKeyInfo, asn1_cryptobyte.SEQUENCE) {
+		return &LintResult{Status: Fatal, Details: "error reading pkixPublicKey"}
+	}
+
+	var algorithm cryptobyte.String
+	if !publicKeyInfo.ReadASN1(&algorithm, asn1_cryptobyte.SEQUENCE) {
+		return &LintResult{Status: Fatal, Details: "error reading pkixPublicKey algorithm"}
+	}
+
+	if !algorithm.SkipASN1(asn1_cryptobyte.OBJECT_IDENTIFIER) {
+		return &LintResult{Status: Fatal, Details: "error reading pkixPublicKey algorithm OID"}
+	}
+
+	if algorithm.Empty() {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier missing required NULL parameter"}
+	}
+
+	var nullValue cryptobyte.String
+	if !algorithm.ReadASN1(&nullValue, asn1_cryptobyte.NULL) {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier with non-NULL parameter"}
+	}
+
+	if len(nullValue) != 0 {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier with NULL parameter containing trailing data"}
+	}
+
+	// ensure algorithm is empty and no trailing data is present
+	if !algorithm.Empty() {
+		return &LintResult{Status: Error, Details: "certificate contains RSA public key algorithm identifier with trailing data"}
+	}
+
+	return &LintResult{Status: Fatal, Details: "certificate rsa algorithm identifier appears correct, but didn't match byte-wise comparison"}
+}
+
+func init() {
+	RegisterLint(&Lint{
+		Name:          "e_rsa_encryption_parameter_not_null",
+		Description:   "RSA: Encoded algorithm identifier MUST have NULL parameters",
+		Citation:      "RFC 4055, Section 1.2",
+		Source:        RFC5280, // RFC4055 is referenced in RFC5280, Section 1
+		EffectiveDate: util.RFC5280Date,
+		Lint:          &rsaEncryptionParamNotNULL{},
+	})
+}
+
+func getOffset(asn1Sequence []byte) int {
+	if len(asn1Sequence) < 2 {
+		return 0
+	}
+
+	if asn1Sequence[1]&0x80 == 0 {
+		// short form length encoding in 1 octet
+		return 2 // 1 tag octet +1 length encoding octet
+	}
+
+	// long form length encoding
+	length := int(asn1Sequence[1] & 0x7f)
```

Comment:
Ditto here RE: adding a brief explanation of the `& 0x7f` operation.

How about tweaking the existing comment to say something like:

"Using long form length encoding. Clear the length encoding indicator in bit 8, the remaining bits contain the length"